File: front/app/src/features/auth/utils/passwordValidator.js
// 비밀번호 정책: 8자 이상 + 영문 + 숫자 + 특수문자 각 1개 이상
// 백엔드 PasswordValidator와 동일 정책 (프론트는 사전 안내용, 최종 검증은 서버)
const PASSWORD_PATTERN = /^(?=.*[A-Za-z])(?=.*\d)(?=.*[!@#$%^&*()_+\-=]).{8,}$/;

export const PASSWORD_GUIDE = '8자 이상, 영문·숫자·특수문자를 포함해주세요.';

/**
 * 비밀번호 정책 검증.
 * @returns {boolean} 정책 통과 여부
 */
export const isValidPassword = (password) => {
  return PASSWORD_PATTERN.test(password ?? '');
};

File: back/app/src/main/java/com/sloway/app/common/util/PasswordValidator.java
package com.sloway.app.common.util;

import com.sloway.app.common.exception.CustomException;
import com.sloway.app.member.common.MemberErrorCode;
import java.util.regex.Pattern;

/**
 * 비밀번호 정책 검증 유틸.
 * 정책: 8자 이상 + 영문 + 숫자 + 특수문자 각 1개 이상.
 * 회원가입/비번변경/비번재설정 등 비번이 설정되는 모든 지점에서 공통 사용.
 */
public final class PasswordValidator {

    // 영문/숫자/특수문자 각 1개 이상 + 8자 이상
    private static final Pattern PASSWORD_PATTERN = Pattern.compile(
            "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[!@#$%^&*()_+\\-=]).{8,}$"
    );

    private PasswordValidator() {} // 인스턴스화 방지

    /**
     * 비밀번호 정책 검증. 위반 시 CustomException.
     */
    public static void validate(String password) {
        if (password == null || !PASSWORD_PATTERN.matcher(password).matches()) {
            throw new CustomException(MemberErrorCode.PASSWORD_TOO_SHORT);
        }
    }
}